### PR TITLE
Serve range requests correctly for static files

### PR DESCRIPTION
**Pull request by tomsmalley**
Without this, Safari will refuse to play video files from /static.

I have:

  - [x] Based work on latest `develop` branch
  - [x] Looked for lint in my changes with `hlint .` (lint found code you did not write can be left alone)
  - [x] Run the test suite: `$(nix-build -A selftest --no-out-link)`
  - [ ] (Optional) Run CI tests locally: `nix-build release.nix -A build.x86_64-linux --no-out-link` (or `x86_64-darwin` on macOS)

### CODE REVIEW

File: lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs [hunk 1]
```diff
@@ -94,8 +95,15 @@ serveAsset' doRedirect base fallback p = do
           let finalFilename = base </> p </> "encodings" </> T.unpack (decodeUtf8 e)
           stat <- liftIO $ getFileStatus finalFilename
           modifyResponse $ setHeader "Last-Modified" "Thu, 1 Jan 1970 00:00:00 GMT"
-          modifyResponse $ setResponseCode 200 . setContentLength (fromIntegral $ fileSize stat) . setContentType (fileType defaultMimeTypes p)
-          sendFile finalFilename
+          modifyResponse $ setHeader "Accept-Ranges" "bytes"
+          modifyResponse $ setContentType (fileType defaultMimeTypes p)
```
Comment (3noch) [inline lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs]: Out of curiosity, why aren't we composing all these `modifyResponse` into one ?

File: lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs [hunk 2]
```diff
@@ -94,8 +95,15 @@ serveAsset' doRedirect base fallback p = do
...
+          let size = fromIntegral $ fileSize stat
+          -- Check if this is a range request
+          req <- getRequest
+          wasRange <- checkRangeReq req finalFilename size
+          unless wasRange $ do
+            modifyResponse $ setResponseCode 200 . setContentLength size
+            sendFile finalFilename
```
Comment (3noch) [inline lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs]: What happens when we *don't* `sendfile`?

File: lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs (hunk 1 above)
Comment (tomsmalley) [reply lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs]: No reason. Will refactor this.

File: lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs (hunk 2 above)
Comment (tomsmalley) [reply lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs]: `checkRangeReq` is a poorly named function which actually does all of the work for range requests.

Comment (3noch) [reply lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs]: Haha I suspected as much... Maybe a comment then.

**Comment by tomsmalley** [comment]
https://github.com/snapframework/snap-core/pull/294
Upstream PR

File: default.nix [hunk 3]
```diff
@@ -42,6 +42,8 @@ let
         hnix-store-core = self.callHackage "hnix-store-core" "0.1.0.0" {};
 
         ghcid = self.callCabal2nix "ghcid" (hackGet ./dep/ghcid) {};
+        # Exports more internals
```
Comment (alexfmpe) [inline default.nix]: Should we put the PR link here?

Comment (3noch) [reply default.nix]: We're tracking this as a separate ticket.

**Pull request merged**